### PR TITLE
MINOR: fix and improve StreamsConfig JavaDocs

### PR DESCRIPTION
**Pull request by mjsax**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java [hunk 1]
```diff
@@ -650,7 +650,7 @@ public void shouldSpecifyCorrectValueSerdeClassOnError() {
     @Test
     public void shouldThrowExceptionIfMaxInFlightRequestsGreaterThanFiveIfEosEnabled() {
         props.put(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, EXACTLY_ONCE);
-        props.put(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION, 7);
+        props.put(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION, 3);
```
Comment (vvcephei) [inline streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java]: Hey @mjsax . Sorry, but I didn't follow this change. How does this test pass when it expects the value to be 7, but it's now 3?

Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java]: Ups -- this change should not be in the PR (I was just trying out locally if 3 is accepted -- the test fails of course for this case).

**Review by vvcephei — APPROVED** [review]
Thanks @mjsax !

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `2.5`

**Pull request merged**